### PR TITLE
Improve `mixer.com` CLI handling of volume and effect arguments

### PR DESCRIPTION
**Pull request by kcgen**
Some invalid input could cause volume to go to infinity. Should be more robust now. For example, decibel and percentages can be mixed, i.e.: `mixer opl 80:d-3`

The reverb and chorus engines will be activated on-demand, i.e: `mixer.com sb r30 c10`, even if the `[mixer]`'s reverb and chorus were off at the time of startup. 

Suggest reviewing commit-by-commit.

### CODE REVIEW

File: src/hardware/mixer.cpp [hunk 1]
```diff
@@ -2123,6 +2123,44 @@ static void SDLCALL MIXER_CallBack([[maybe_unused]] void *userdata,
...
+		const auto l = to_volume(parts[0]);
+		const auto r = to_volume(parts[1]);
+		if (l && r) {
+			AudioFrame volume = {*l, *r};
+			return volume;
+		}
+	}
+	return {};
+}
+
```
Comment (johnnovak) [inline src/hardware/mixer.cpp:2171]: Very nice improvement, much cleaner and more robust 💯

File: src/hardware/mixer.cpp [hunk 2]
```diff
@@ -2354,10 +2354,10 @@ class MIXER final : public Program {
 		                        const std::string &chorus) {
 			WriteOut(column_layout.c_str(),
 			         name.c_str(),
-			         volume.left * 100.0f,
-			         volume.right * 100.0f,
-			         gain_to_decibel(volume.left),
-			         gain_to_decibel(volume.right),
+			         static_cast<double>(volume.left * 100.0f),
+			         static_cast<double>(volume.right * 100.0f),
```
Comment (johnnovak) [inline src/hardware/mixer.cpp:2358]: `gain_to_percentage` to improve readability?

**Review by johnnovak — APPROVED** [review]
Great work @kcgen, just a few minor comments. (I've only reviewed this on my phone, haven't returned to Australia yet.)

File: src/hardware/mixer.cpp (hunk 1 above)
Comment (kcgen) [reply src/hardware/mixer.cpp:2171]: Thanks for the review, @johnnovak !

File: src/misc/string_utils.cpp [hunk 3]
```diff
@@ -233,3 +233,41 @@ void clear_language_if_default(std::string &l)
...
 	}
 }
+
+// Parse a percentage in string form and return it as a float ie: "100"
+// -> 100.0f, 10 -> 10.0f. The value can be bound using optional arguments.
+std::optional<float> parse_value(const std::string &s, const float min_value, const float max_value)
+{
+	try {
+		return std::clamp(std::stof(s), min_value, max_value);
+	} catch (...) {
```
[the remark this answers is not in the document]
Comment (kcgen) [reply src/misc/string_utils.cpp:243]: Added

File: src/hardware/mixer.cpp (hunk 2 above)
Comment (kcgen) [reply src/hardware/mixer.cpp:2358]: Also renamed `percent_to_ratio` to `percentage_to_gain` to follow the pattern.

**Pull request merged**